packages/viewer/src/interactions/flyToPartKeyInteraction.ts

```diff
@@ -0,0 +1,37 @@
+import { StreamApi, toProtoDuration } from '@vertexvis/stream-api';
+import { KeyInteraction, KeyState } from './keyInteraction';
+import { ConfigProvider } from '../config/config';
+import { TapEventDetails } from './tapEventDetails';
+
+export class FlyToPartKeyInteraction
+  implements KeyInteraction<TapEventDetails> {
+  public constructor(
+    private stream: StreamApi,
+    private configProvider: ConfigProvider
+  ) {}
+
+  public predicate(keyState: KeyState): boolean {
+    return keyState['Meta'] || keyState['Control'];
+  }
+
+  public async fn(e: TapEventDetails): Promise<void> {
+    const hitResult = await this.stream.hitItems(
+      {
+        point: e.position,
+      },
+      true
+    );
+
+    if (
+      hitResult.hitItems?.hits != null &&
+      hitResult.hitItems.hits.length > 0
+    ) {
+      await this.stream.flyTo({
+        itemId: hitResult.hitItems.hits[0].itemId,
+        animation: {
+          duration: toProtoDuration(this.configProvider().animation.durationMs),
+        },
+      });
+    }
```

Comment:
Since the user will be expecting an action here from the sdk given that this interaction is turned on, should we add a warning or log or something that no parts were found? Not necessary, just a thought.

Reply:
I guess my thought on that would be that I'd expect a click in empty space to do nothing with this interaction, and the warning might get a bit noisy (granted I don't expect people would be Command+clicking in white space often). Maybe a debug log would serve the same purpose and communicate it's expected behavior?